servant/src/Servant/API/ContentTypes.hs

```diff
@@ -307,6 +304,28 @@ instance MimeRender OctetStream ByteString where
 instance MimeRender OctetStream BS.ByteString where
     mimeRender _ = fromStrict
 
+-- | A type for responses with content-body.
+data NoContent = NoContent
+  deriving (Show, Eq, Read)
+
+instance FromJSON NoContent where
+    parseJSON _ = return NoContent
```

Comment:
We should have a test that makes sure that `aeson` doesn't crash before it calls `parseJSON`. In the `NoContent` case we don't have an input that's a valid json value, so `aeson` cannot construct the `Value` that it would pass to `parseJSON`. (Hopefully laziness saves us here, but we need a test.)